### PR TITLE
Feat/coupon modal UI

### PR DESCRIPTION
**Pull request by femacedogomes**

### CODE REVIEW

**Comment by femacedogomes** [comment]
WAITING generate coupon code endpoint

File: src/components/orgs/SupplierCoupons/index.tsx [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+      .then((res) => {
+        setCouponCode(res?.data?.coupon?.code);
+        setTimeout(() => {
+          handleNextStep();
+          setTimeout(() => {
+            handleNextStep();
+          }, 1500);
+        }, 3000);
+      })
+      .then()
```
Comment (renanpontez) [inline src/components/orgs/SupplierCoupons/index.tsx:148]: Why this?

File: src/components/orgs/SupplierCoupons/index.tsx [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+  icon,
+  couponTitle,
+}) => {
+  const [couponCode, setCouponCode] = useState("");
+  const [showCouponModal, setShowCouponModal] = useState(false);
+  const [currentStep, setCurrentStep] = useState(0);
+
+  const handleNextStep = () => {
+    setCurrentStep((prev) => prev + 1);
+  };
```
Comment (renanpontez) [inline src/components/orgs/SupplierCoupons/index.tsx:47]:
```suggestion
  const handleNextStep = (step) => {
    setCurrentStep(step);
  };
```

Comment (renanpontez) [inline src/components/orgs/SupplierCoupons/index.tsx:47]:
```suggestion
  const handleNextStep = (step) => setCurrentStep(step);
```

File: src/components/orgs/SupplierCoupons/index.tsx [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+          className="w-full mx-4 max-w-md !bg-white !text-primary-main !shadow-none"
+          onClick={() => setShowCouponModal(false)}
+        >
+          voltar
+        </ButtonPrimary>
+      </div>
+    );
+  };
+
+  const ActiveCounpon = async () => {
```
Comment (renanpontez) [inline src/components/orgs/SupplierCoupons/index.tsx:132]:
```suggestion
  const ActiveCoupon = async () => {
```

File: src/components/orgs/SupplierCoupons/index.tsx [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+    const res: any = await await couponsService.generateCouponCode(id);
```
Comment (renanpontez) [inline src/components/orgs/SupplierCoupons/index.tsx:133]: Why `any`?

File: src/components/atoms/NcModal/NcModal.tsx [hunk 5]
```diff
@@ -25,6 +26,7 @@ const NcModal: FC<NcModalProps> = ({
   modalTitle = "Modal title",
   isOpenProp,
   onCloseModal,
+  noCloseButton = false,
```
Comment (renanpontez) [inline src/components/atoms/NcModal/NcModal.tsx:29]: noCloseButton ou onCloseButton?

File: src/components/mols/Coupon/index.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        </div>
+        <AccordionInfo
+          data={[
+            {
+              name: "Regras de uso",
+              content:
+                "Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsa, dolorem. Quae debitis hic ipsum inventore assumenda laborum reprehenderit, asperiores nostrum, molestiae odio excepturi maiores possimus ad cum quia et doloremque?",
+            },
+          ]}
+        ></AccordionInfo>
```
Comment (renanpontez) [inline src/components/mols/Coupon/index.tsx:88]:
```suggestion
        <AccordionInfo
          data={[
            {
              name: "Regras de uso",
              content:
                "Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsa, dolorem. Quae debitis hic ipsum inventore assumenda laborum reprehenderit, asperiores nostrum, molestiae odio excepturi maiores possimus ad cum quia et doloremque?",
            },
          ]}
       />
```

File: src/components/mols/CouponActived/index.tsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  couponActivateCode: string;
+  couponTitle: string;
+  expirateTime: string;
+  supplierLogo: string | StaticImageData;
+  supplierName: string;
+  couponDiscount: string;
+  supplierCategory: string;
+}
+
+const CouponActivedPage: React.FC<CouponProps> = ({
```
Comment (renanpontez) [inline src/components/mols/CouponActived/index.tsx:19]:
```suggestion
const CouponActivatedPage: React.FC<CouponProps> = ({
```

File: src/components/mols/CouponActived/index.tsx [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        </div>
+        <AccordionInfo
+          data={[
+            {
+              name: "Regras de uso",
+              content:
+                "Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsa,",
+            },
+          ]}
+        ></AccordionInfo>
```
Comment (renanpontez) [inline src/components/mols/CouponActived/index.tsx:86]:
```suggestion
        <AccordionInfo
          data={[
            {
              name: "Regras de uso",
              content:
                "Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsa,",
            },
          ]}
        />
```

File: src/components/mols/Modal/ModalDelete.tsx [hunk 9]
```diff
@@ -6,9 +6,16 @@ import NcModal from "@/components/atoms/NcModal/NcModal";
...
+  closeOnBlur?: boolean;
+  noCloseButton?: boolean;
 }
 
-const ModalDelete: FC<ModalDeleteProps> = ({ show, onCloseModalDelete }) => {
+const ModalDelete: FC<ModalDeleteProps> = ({
+  show,
+  onCloseModalDelete,
+  closeOnBlur = true,
+  noCloseButton = false,
```
Comment (renanpontez) [inline src/components/mols/Modal/ModalDelete.tsx:17]: noCloseButton ou onCloseButton?

File: src/components/orgs/CouponListPage/index.tsx [hunk 10]
```diff
@@ -125,10 +121,18 @@ const CouponListPage: React.FC<ICouponListPageProps> = ({ supplierId }) => {
           supplier?.coupons.length > 0 ? (
             supplier?.coupons.map((coupon: ICoupon, key) => (
               <SupplierCoupons
-                icon={supplier.supplier?.id == session?.user.id ? Edit : Arrow}
+                couponTitle={coupon.title}
+                icon={supplier.supplier.id == session?.user.id ? Edit : Arrow}
```
Comment (renanpontez) [inline src/components/orgs/CouponListPage/index.tsx:125]:
```suggestion
                icon={supplier.supplier.id === session?.user.id ? Edit : Arrow}
```

From now on, stop using simple comparison. **Always use `===` and `!==`**

File: src/components/orgs/SupplierCoupons/index.tsx (hunk 1 above)
Comment (femacedogomes) [reply src/components/orgs/SupplierCoupons/index.tsx:148]: sorry, i forgot to remove..

File: src/hooks/useDateDifferenceInDays.tsx [hunk 11]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+
+const useDateDiffInDays = (date: string) => {
```
Comment (pedrohbl03) [inline src/hooks/useDateDifferenceInDays.tsx:3]: this is a HOC. 

High Order Component !== Hooks

hooks encapsulate a business rule

and HOCs (High Order Components) are intelligent components

Comment (pedrohbl03) [reply src/hooks/useDateDifferenceInDays.tsx:3]: change this to a atoms folder and remove 'use' from ur name.

Comment (pedrohbl03) [reply src/hooks/useDateDifferenceInDays.tsx:3]: and u can use ''dayjs" here too, to make a more improve logic. 

example: https://day.js.org/docs/en/display/difference

**Review by pedrohbl03 — CHANGES_REQUESTED** [review]

File: src/components/atoms/NcModal/NcModal.tsx (hunk 5 above)
Comment (femacedogomes) [reply src/components/atoms/NcModal/NcModal.tsx:29]: its noCloseButton, to remove the Button on top right. But i will change to hasCloseButton.

File: src/components/atoms/DateDifferenceInDays/index.tsx [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import dayjs from "dayjs";
+
+const DateDifferenceInDays = (date: string) => {
```
Comment (renanpontez) [inline src/components/atoms/DateDifferenceInDays/index.tsx:4]:
```suggestion
const getDateDiffInDays = (date: string) => {
```

File: src/components/atoms/DateDifferenceInDays/index.tsx [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+  const today = dayjs();
+  const validity = dayjs(date);
+
+  const DateDiff = validity.diff(today, "day");
```
Comment (renanpontez) [inline src/components/atoms/DateDifferenceInDays/index.tsx:8]:
```suggestion
  const dateDiff = validity.diff(today, "day");
```

**Review by pedrohbl03 — APPROVED** [review]

**Review by renanpontez — APPROVED** [review]
Commented minor changes. LGTM

**Pull request merged**